### PR TITLE
Add The Mobile Menu Functionality using Javascript

### PR DESCRIPTION
**Pull request by Abdulwhbhussain**
**Add the JS Linters;
Add the main.js File;
Link index.html to main.js File;
Write Event Listeners and Handlers in Main.js to put Menu Functionality.**

### CODE REVIEW

**Comment by huseyinbiyik** [comment]
Hi @WHB901 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

Alt Text

## Invalid Code Review Request

⚠️ You have submitted a project, that has no visible contribution from your pair programming partner. Kindly check [Work and Submission mode](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/7_mobile_menu.md#work-and-submission-mode).


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: index.html [hunk 1]
```diff
@@ -419,5 +427,13 @@ <h2 class="form-header">Contact me</h2>
...
       </section>
     </div>
+    <div class="overlay-menu hidden">
+      
+      <a href="#portfolio" style="display: block; color: white;">Portfolio</a>
+            <a href="#about" style="display: block; color: white;">About</a>
+            <a href="#contact" style="display: block; color: white;">Contact</a>
+    </div>
+
+    <script src="./main.js"></script>
```
Comment (ShadyShawkat) [inline index.html:437]: - Kindly make sure to match the Figma design as much as possible, which includes:
   - [ ] Using the proper font size indicated in the original design.
   - [ ] Implementing a "close button" on the top right corner of the mobile menu which functions to close the menu as per the design.
   - [ ] Editing the spacing between and around the mobile menu links to match the design.

| Your design | Original design |
| --- | --- |
| image | image |

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 1 above)
Comment (Abdulwhbhussain) [reply index.html]: Fixed The Issue.

File: index.html [hunk 2]
```diff
@@ -419,5 +427,13 @@ <h2 class="form-header">Contact me</h2>
...
       </section>
     </div>
+    <div class="overlay-menu hidden">
+      <a><bold>&times;</bold></a>
+      <a href="#portfolio" style="display: block; color: white;"><b>Portfolio</b></a>
+            <a href="#about" style="display: block; color: white;"><b>About</b></a>
+            <a href="#contact" style="display: block; color: white;"><b>Contact</b></a>
+    </div>
+
+    <script src="./main.js"></script>
```
Comment (Shedrack-Sunday) [inline index.html:437]: - [ ] Good job making most of the changes highlighted by the previous reviewer 👏 . However, there's one left, which happens to be the close button.
- I could cancel the menu pop up by clicking the empty space by the top right corner, but its not user diecting and does not follow requiements. Kindly add button, which might be a case of `CSS`, to give a better user experince 👍 

## App view:
image

## Figma:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @Abdulwhbhussain            👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 


## Highlights
- Correct use of GitHub Flow ✔
- Well-structured files ✔
- All linter checks are passing ✔
- Moderate design  ✔
  
# Required Changes ♻️

Check the comments under the review.
You just have to make a few changes based on project requirements, and you're on your way 🚀. 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Trast00 — APPROVED** [review]
# Approved 🟢
Hi @Abdulwhbhussain , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
allfather-approves-approved

## Highlights
- Project look awsome 🍏 🌷 
- Code is clean and well organized 🧼 💯 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@Trast00) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**